packages/launcher/src/renderer/launcher/contacts/ContactsScreen.js

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable no-console */
+// @flow
+import React, { Component } from 'react'
+import {
+  graphql,
+  createFragmentContainer,
+  commitMutation,
+  QueryRenderer,
+} from 'react-relay'
+
+import { EnvironmentContext } from '../RelayEnvironment'
+import LauncherContext from '../LauncherContext'
+import RelayLoaderView from '../RelayLoaderView'
+import ContactsView, {
+  type Contact,
+  type SubmitContactInput,
+} from './ContactsView'
+
+type QueryProps = {
+  userID: string,
+}
+
+type Props = QueryProps & {
+  contacts: {
+    userContacts: Array<Contact>,
+  },
+}
+
+type State = {
+  data: Array<Object>,
+}
+
+export const addContactMutation = graphql`
+  mutation ContactsScreenAddContactMutation(
+    $input: AddContactInput!
+    $userID: String!
+  ) {
+    addContact(input: $input) {
+      viewer {
+        contacts {
+          ...ContactsView_contacts @arguments(userID: $userID)
+        }
+      }
+    }
+  }
+`
+
+export class ContactsScreen extends Component<Props, State> {
+  static contextType = EnvironmentContext
+  state = {
+    data: [],
+  }
+
+  addNewContact = (data: SubmitContactInput): Promise<Contact> => {
```

Comment:
Does this need to be a promise returning the `Contact`? The mutation result should update the UI, no?